ParkingZoneApp/Areas/Identity/Pages/Account/Manage/Index.cshtml.cs

```diff
@@ -98,18 +99,31 @@ public async Task<IActionResult> OnPostAsync()
                 await LoadAsync(user);
                 return Page();
             }
-
             var phoneNumber = await _userManager.GetPhoneNumberAsync(user);
+
             if (Input.PhoneNumber != phoneNumber)
             {
                 var setPhoneResult = await _userManager.SetPhoneNumberAsync(user, Input.PhoneNumber);
+
                 if (!setPhoneResult.Succeeded)
                 {
                     StatusMessage = "Unexpected error when trying to set phone number.";
                     return RedirectToPage();
                 }
             }
 
+            if (Input.Name != user.Name)
+            {
+                user.Name = Input.Name;
+                user.UserName = Input.Name;
```

Comment:
Why after changing Nam, UserName is being equal to Name.
UserName never changes. After registration UserName automatically gets Email.